pliers/extractors/text.py

```diff
@@ -402,3 +400,140 @@ def _extract(self, stim):
+            for further info.
+    '''
+
+    _log_attributes = ('pretrained_model', 'encoding_level',
+                       'pooling', 'framework', 'tokenizer_type',
+                       'model_kwargs', 'tokenizer_kwargs')
+
+    def __init__(self,
+                 pretrained_model_or_path='bert-base-uncased',
+                 tokenizer='bert-base-uncased',
+                 framework='pt',
+                 encoding_level='token',
+                 pooling=None,
+                 model_kwargs=None,
+                 tokenizer_kwargs=None):
+                 
+        verify_dependencies(['transformers'])
+
+        if framework not in ['pt', 'tf']:
+            raise(ValueError('''Invalid framework;
+                must be one of 'pt' (pytorch) or 'tf' (tensorflow)'''))
+
+        if model_kwargs is None:
+            model_kwargs = {}
+
+        if tokenizer_kwargs is None:
+            tokenizer_kwargs = {}
+
+        self.pretrained_model = pretrained_model_or_path
+        self.tokenizer_type = tokenizer
+        self.framework = framework
+        self.encoding_level = encoding_level
+        self.pooling = pooling
+        self.model_kwargs = model_kwargs
+        self.tokenizer_kwargs = tokenizer_kwargs
+        model_name = 'BertModel' if self.framework == 'pt' else 'TFBertModel'
+
+        self.tokenizer = transformers.BertTokenizer.from_pretrained(
+            tokenizer, **tokenizer_kwargs)
+        self.model = getattr(transformers, model_name).from_pretrained(
+            pretrained_model_or_path, **model_kwargs)
+
+        super(PretrainedBertEncodingExtractor, self).__init__()
+
+    def _extract(self, stims):
+
+        stims.name = ' '.join([s.text for s in stims.elements])
+        text, onsets, durations = zip(
+            *((s.text, s.onset, s.duration) for s in stims.elements))
+        durations = [0.0 if d is None else d for d in durations]
+
+        tokens = [self.tokenizer.tokenize(t) for t in text]
+        tokens_flat = list(flatten(tokens))
+
+        def cast_to_token_level(word_level_list):
+            token_level_list = [listify(word_level_list[i]) * len(tok)
+                                for i, tok in enumerate(tokens)]
+            token_level_list = list(flatten(token_level_list))
+            return token_level_list
+
+        t_text, t_ons, t_dur = map(cast_to_token_level,
+                                   [text, onsets, durations])
+
+        tensor_tokens = self.tokenizer.encode(tokens_flat,
+                                              return_tensors=self.framework)
+        output = self.model(tensor_tokens)
+        output = [out.detach().numpy() if self.framework == 'pt' else
+                  out.numpy() for out in output]
+
+        if self.encoding_level == 'token':
+            encoded_tokens = tokens_flat
+            encodings = output[0][:, 1:-1, :].squeeze()
+
+        elif self.encoding_level == 'sequence':
```

Comment:
What happens if `self.encoding_level` isn't either `token` or `sequence`? I assume this is invalid, in which case there should probably be a validation check at the very start of the `__init__` that raises a `ValueError` if `encoding_level not in {'token', 'sequence'}`. Otherwise we could do a lot of loading/computation only to fail here.

Reply:
added `ValueError`